src/kernel/arch/x86/syscalls.zig

```diff
@@ -0,0 +1,248 @@
+const arch = @import("arch.zig");
+const testing = @import("std").testing;
+const assert = @import("std").debug.assert;
+const isr = @import("isr.zig");
+const log = @import("../../log.zig");
+
+pub const INTERRUPT: u16 = 0x80;
+pub const NUM_HANDLERS: u16 = 256;
+
+/// A syscall handler
+pub const SyscallHandler = fn (ctx: *arch.InterruptContext) u32;
```

Comment:
I have seen you are a bit inconsistent in the formatting of types where you have spaces and sometimes not. We will need to add this to the formatting rules. Personally, I would go with having the spaces, easier to read.

Reply:
Hmm, yeah I agree on the spaces. Let me know where I've missed spaces before and I'll fix it. Zig fmt will probably notice it though :)